README.md

```diff
@@ -157,10 +157,22 @@ Access to the security namespace and istio-system should be highly restricted.
 
 ## Install Istio-CNI
 
-TODO. This is an optional step - CNI must run in a dedicated namespace, it is a 'singleton' and extremely 
+This is an optional step - CNI must run in a dedicated namespace, it is a 'singleton' and extremely 
 security sensitive. Access to the CNI namespace must be highly restricted. 
 
-It is possible to add Istio-CNI later, and gradually migrate. 
+**NOTE:** The environment variable `ISTIO_CLUSTER_ISGKE` is assumed to be set to `true` if the cluster
+is a GKE cluster.
+
+```bash
+ISTIO_CNI_ARGS=
```

Comment:
Can we determine this from kubectl cluster-info ? If yes, we can add it to the iop script ( and the future go binary that will do the same thing ). 

Also - wondering if there is any detection that can be done in cni itself...

(for future PR)

Reply:
I'll see what I can figure out with cluster-info.  The problem with trying to do it in the CNI itself is that the `istio-cni-node` pod needs to volume mount the host's cni-bin-dir so it can copy in the Istio CNI binary.  So, to discover the cni-bin-dir or which cloud-platform we're on we'd need to have something like a k8s operator for CNI.

Reply:
Can we mount both ? With optional=true maybe ? Anything we can do to simplify the UX.
But we can do it in separate PRs - it's fine to use a values.yaml setting for now - doesn't need to be 
part of the iop script.

Reply:
AFAIK, there's no "optional" setting for host mounts.  We should be able to use "DirectoryOrCreate" and check in the install script if the directory is empty or not before installing the binary to detect.  Will create a CNI PR for that shortly.

Reply:
Got it, thanks for confirming.

For a separate PR maybe we can add some detection (kubect cluster info or something)